textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/DataReader.java

```diff
@@ -59,6 +72,26 @@ public void open() throws DataFlowException {
             TopDocs topDocs = luceneIndexSearcher.search(dataReaderPredicate.getLuceneQuery(), Integer.MAX_VALUE);
             scoreDocs = topDocs.scoreDocs;
             cursor = OPENED;
+
+            this.queryTokens = Utils.tokenizeQuery(dataReaderPredicate.getAnalyzer(),dataReaderPredicate.getQueryString());
+
+            // sort the query tokens, as the term vector are also sorted.
+            // This makes the seek faster.
+            this.queryTokens.sort(String.CASE_INSENSITIVE_ORDER);
+
+            // The terms in the term vector are stored as ByteRef,

```

Comment:
I think I need to be educated here.  What's the advantage of using ByteRef instead of a String for each query token?